lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOActivityDAOImpl.java

```diff
@@ -111,7 +112,11 @@ public Activity get(String id) throws DatastoreException, NotFoundException {
 	public void delete(String id) throws DatastoreException {
 		MapSqlParameterSource param = new MapSqlParameterSource();
 		param.addValue(COL_ACTIVITY_ID.toLowerCase(), id);
-		basicDao.deleteObjectByPrimaryKey(DBOActivity.class, param);
+		try {
+			basicDao.deleteObjectByPrimaryKey(DBOActivity.class, param);
+		} catch (DataIntegrityViolationException e) {
+			throw new IllegalArgumentException("Cannot delete activity "+id+". This can happen if some entity is 'generated by' the activity.", e);
```

Comment:
Give user hint about how to avoid the error. "If you wish to delete this activity, please first delete all Entities generated by this Activity."